cmd/kube-controller-manager/app/core.go

```diff
@@ -178,6 +178,7 @@ func startEndpointController(ctx ControllerContext) (bool, error) {
 	go endpointcontroller.NewEndpointController(
 		ctx.InformerFactory.Core().V1().Pods(),
 		ctx.InformerFactory.Core().V1().Services(),
+		ctx.InformerFactory.Core().V1().Endpoints(),
```

Comment:
I didn't realize we weren't watching endpoints before - given that this controller is expected to know and mutate every endpoint, I think this is exactly what we should be doing.  It will have a memory impact, but will drop a significant chunk of QPS from the apiservers.

@wojtek-t @gmarek may change our kube-mark profiles, but I would expect it to be in a positive way.